File: Problems/Algorithms/14. Longest Common Prefix/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        prefix = strs[0]
        for i in range(1, len(strs)):
            while True:
                try:
                    if strs[i].index(prefix) != 0:
                        prefix = prefix[:-1]
                    else:
                        break
                except ValueError:
                    prefix = prefix[:-1]
            if not prefix:
                return ""
        return prefix

File: Problems/Algorithms/14. Longest Common Prefix/longest_common_prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
        match strs.is_empty() {
            true => "".to_string(),
            false => strs.iter().skip(1).fold(strs[0].clone(), |acc, x| {
                acc.chars()
                    .zip(x.chars())
                    .take_while(|(x, y)| x == y)
                    .map(|(x, _)| x)
                    .collect()
            }),
        }
    }
}

